src/sqlfluff/core/parser/segments/raw.py

```diff
@@ -148,3 +151,51 @@ def get_end_pos_marker(self):
     def get_start_pos_marker(self):
         """Return the pos marker at the start of this segment."""
         return self.pos_marker
+
+
+class RawSegmentFactory:
+    optional = False
+    _anti_template = None
```

Comment:
Why are these two left as class attributes? I'm surprised that `optional` is here but `_template` is an instance attribute. In my head they're the same kind of property.

Reply:
This is a shortcut sometimes used in Python -- attributes on a class can be accessed using `self.whatever`, so they act a bit like default-valued fields. Specific instances of the class can "override" them by storing them as fields instead. This pattern is not supported by statically-typed languages like C++ or Java, so it's not well known. These could be moved to `__init__()` and treated like normal instance fields.